packages/zoe/src/contractFacet/zcfSeat.js

```diff
@@ -28,24 +34,76 @@ export const createSeatManager = (zoeInstanceAdmin, getAssetKindByBrand) => {
     assert(activeZCFSeats.has(zcfSeat), X`seat has been exited`);
   };
 
+  /**
+   * @param {ZCFSeat} zcfSeat
+   * @returns {void}
+   */
   const doExitSeat = zcfSeat => {
     assertActive(zcfSeat);
     activeZCFSeats.delete(zcfSeat);
   };
 
+  /**
+   * @param {ZCFSeat} zcfSeat
+   * @returns {Allocation}
+   */
   const getCurrentAllocation = zcfSeat => {
     assertActive(zcfSeat);
     return activeZCFSeats.get(zcfSeat);
   };
 
-  const commitSeatStaging = seatStaging => {
+  /**
+   * @param {ZCFSeat} zcfSeat
+   * @returns {void}
+   */
+  const commitStagedAllocation = zcfSeat => {
+    assertActive(zcfSeat);
+    activeZCFSeats.set(zcfSeat, zcfSeat.getStagedAllocation());
+    zcfSeatToStagedAllocations.delete(zcfSeat);
```

Comment:
The `delete` on line 62 seems exactly the same as the `set` on line 61. It should be impossible to get here if `zcfSeat` is not in `zcfSeatToStagedAllocations`. If that is not already obviously impossible, it must be made obviously impossible. If it happens anyway we reliable get a low level failure from the `delete`, which throws within the atomic action. Once I use the atomic action API to ensure that such failures cause immediate vat failure, we're safe even against these specific cannot-happen conditions.

When I will ask you to review the PR that protects such atomic actions, please keep these "impossible" cases in mind.